pkg/controller/backup/backup_controller.go

```diff
@@ -263,3 +295,229 @@ func (c *Controller) enqueueBackup(obj interface{}) {
 	}
 	c.queue.Add(key)
 }
+
+func (c *Controller) isAlreadyRecordfailure(backup *v1alpha1.Backup) bool {
```

Comment:
All the following logic is for Snapshot failure retry, can you put them in a separate object but not Controller-wide since it's not shared?

Reply:
the same as https://github.com/pingcap/tidb-operator/pull/4895#discussion_r1119522188, other backup is treated as special, will not record their failure and mard as failed directly.